src/librustdoc/html/static/main.js

```diff
@@ -2332,12 +2332,16 @@ function defocusSearchBar() {
             var dontApplyBlockRule = toggle.parentNode.parentNode.id !== "main";
             if (action === "show") {
                 removeClass(relatedDoc, "fns-now-collapsed");
-                removeClass(docblock, "hidden-by-usual-hider");
+                if (hasClass(docblock, "stability") === false) {
+                    removeClass(docblock, "hidden-by-usual-hider");
+                }
                 onEachLazy(toggle.childNodes, adjustToggle(false, dontApplyBlockRule));
                 onEachLazy(relatedDoc.childNodes, implHider(false, dontApplyBlockRule));
             } else if (action === "hide") {
                 addClass(relatedDoc, "fns-now-collapsed");
-                addClass(docblock, "hidden-by-usual-hider");
+                if (hasClass(docblock, "stability") === false) {
+                    addClass(docblock, "hidden-by-usual-hider");
+                }
```

Comment:
What does stability have to do with it? IIUC this is hiding it when you have `#[unstable]` in the source - why?

Reply:
Oh that's interesting in fact! Let me check that. If you're right, we're badly handling a few things in there.

Reply:
Ok, so "stability" groups a few things, including the feature attributes. Inside it, it is the "portability" class the important one. However, we don't generate items like "deprecated" on implementations as it seems. And even if we did, it'd make sense to print them, and since they're stored inside the `stability` element... For reference, it looks like this:

```html
<div class="stability">
  <div class="stab portability">This is supported on <strong>non-crate feature <code>traits</code></strong> only.</div>
</div>
```

And deprecated looks like this:

```html
<div class="stability">
  <div class="stab deprecated"><span class="emoji">👎</span> Deprecated</div>
</div>
```

And when you merge both:

```html
<div class="stability">
  <div class="stab deprecated"><span class="emoji">👎</span> Deprecated</div>
  <div class="stab portability">This is supported on <strong>non-crate feature <code>traits</code></strong> only.</div>
</div>
```

So it kinda makes sense, however we should maybe rename the item class. What do you think about renaming it then? However, I'll do it in another PR to prevent mixing too much things.

Reply:
Printing both even on implementations sounds like the right approach to me :) happy to wait for another cleanup PR.